daffodil-cli/src/main/scala/org/apache/daffodil/cli/Main.scala

```diff
@@ -226,14 +226,30 @@ class CLIConf(arguments: Array[String], stdout: PrintStream, stderr: PrintStream
   implicit def rootNSConverter = org.rogach.scallop.singleArgConverter[RefQName](qnameConvert _)
 
   implicit def fileResourceURIConverter = singleArgConverter[URI]((s: String) => {
-    val file = new File(s)
-    val uri =
-      if (file.isFile()) {
-        Some(file.toURI)
-      } else {
-        Misc.getResourceRelativeOption(s, None)
+    val optURI = None
+      .orElse {
+        val file = new File(s)
+        if (file.isFile) Some(file.toURI)
+        else None
+      }
+      .orElse {
+        if (s.startsWith("/")) {
+          val resource = this.getClass.getResource(s)
+          Option(resource.toURI)
+        } else {
+          None
+        }
+      }
+      .orElse {
+        val resource = this.getClass.getResource("/" + s)
+        if (resource != null) {
+          Logger.log.warn(s"Found relative path on classpath absolutely, did you mean /$s")
+          Some(resource.toURI)
```

Comment:
If the previous `.orElse` code returns None because a resource cannot be found when s starts with "/", this code will add another "/" to s, still fail to find the resource, and return None again.

Should we combine these two `.orElse` cases by assigning an if expression to `val resource`?

```scala
   val resource = if (s.startsWith("/")) s else "/" + s
```

Actually, you want to warn the user if a relative path is found absolutely on the classpath, but you can guard the warning with an `if (!s.startsWith("/"))`.

Line 247 doesn't seem to be covered by tests too.

Reply:
My thinking was that when we want to remove this fallback support and result in a normal "could not find schema" error, then we can just remove this orElse block and nothing else needs to be modified. I don't feel strongly about this though, I can merge the logic.